WordPress/Classes/Services/SuggestionService.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+
+extension NSNotification.Name {
+    static let suggestionListUpdated = NSNotification.Name("SuggestionListUpdated")
+}
+
+@objc extension NSNotification {
+    public static let suggestionListUpdated = NSNotification.Name.suggestionListUpdated
+}
+
+@objc public class SuggestionService: NSObject {
+
+    // NSCache works with classes such as NSArray, not structs such as Suggestion or [Suggestion].
```

Comment:
Suggestion currently isn't a struct, though?

Reply:
Thanks for catching! (It was at one point during my refactor, then I switched to a class.)
I'll see if we can switch it back over to a struct using the approach you suggested here: https://github.com/wordpress-mobile/WordPress-iOS/pull/14888#pullrequestreview-493611751

Reply:
> I'll see if we can switch it back over to a struct using the approach you suggested here: #14888 (review)

It wasn't possible to make a straightforward switch over to structs, but I did get rid of some of the `@objc` attributes.